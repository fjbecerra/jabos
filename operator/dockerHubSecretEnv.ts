import { EnvVar } from './imports/k8s';

export default function (dockerHub): EnvVar[] {
  return !dockerHub ? [] : [
    {
      name: "DOCKER_HUB_USERNAME",
      valueFrom: {
        secretKeyRef: {
          name: dockerHub.secret,
          key: dockerHub.username
        }
      }
    },
    {
      name: "DOCKER_HUB_PASSWORD",
      valueFrom: {
        secretKeyRef: {
          name: dockerHub.secret,
          key: dockerHub.password
        }
      }
    }
  ];
}
