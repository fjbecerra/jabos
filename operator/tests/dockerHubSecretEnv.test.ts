import dockerHubSecretEnv from '../dockerHubSecretEnv';
import { setMock, clearMock } from './settingsMock';

beforeEach(() => {
  setMock();
});

afterEach(() => {
  clearMock();
});

test('dockerHubSecretEnv without input', () => {
  expect(dockerHubSecretEnv(undefined)).toMatchSnapshot();
});

test('dockerHubSecretEnv with input', () => {
  const dockerHub = {
    secret: 'secret_value',
    username: 'username_value',
    password: 'password_value'
  }
  expect(dockerHubSecretEnv(dockerHub)).toMatchSnapshot();
});
