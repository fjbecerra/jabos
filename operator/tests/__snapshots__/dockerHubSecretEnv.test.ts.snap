// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dockerHubSecretEnv with input 1`] = `
Array [
  Object {
    "name": "DOCKER_HUB_USERNAME",
    "valueFrom": Object {
      "secretKeyRef": Object {
        "key": "username_value",
        "name": "secret_value",
      },
    },
  },
  Object {
    "name": "DOCKER_HUB_PASSWORD",
    "valueFrom": Object {
      "secretKeyRef": Object {
        "key": "password_value",
        "name": "secret_value",
      },
    },
  },
]
`;

exports[`dockerHubSecretEnv without input 1`] = `Array []`;
